### PR TITLE
Allow :format "none" to be specified by clients

### PR DESCRIPTION
**Pull request by clumsyjedi**
so that client software which cannot interpret clojure can still
interact with the bencode data

### CODE REVIEW

File: src/refactor_nrepl/middleware.clj [hunk 1]
```diff
@@ -35,6 +35,11 @@
   `(with-errors-being-passed-on ~transport ~msg
      (transport/send ~transport (response-for ~msg ~(apply hash-map kv)))))
 
+(defn format-response [{:keys [format] :as msg} response]
```
Comment (expez) [inline src/refactor_nrepl/middleware.clj]: I think a better name here would be `serialize-response` or something along those lines.  To me `format-response` makes me think about pretty printing more than producing different data formats.

File: src/refactor_nrepl/middleware.clj [hunk 2]
```diff
@@ -35,6 +35,11 @@
...
+  (if (= format "none")
```
Comment (expez) [inline src/refactor_nrepl/middleware.clj]: I think we should switch this around and do `(= format "edn")` so people aren't left wondering what exactly `none` means.

Comment (bbatsov) [inline src/refactor_nrepl/middleware.clj]: Yep, "none" is pretty confusing IMO.

**Comment by expez** [comment]
There's also a capitalization mistake in the commit message that you might fix while you're addressing the other remarks.

I think other people might be interested in this, so adding a note about this in the `README` might not be such a bad idea.  If you do that, you should also write a test to ensure we don't break this.

I'm OK with leaving this as an easteregg, though, since any bugs will be filed over in your repo :)

File: src/refactor_nrepl/middleware.clj [hunk 3]
```diff
@@ -35,6 +35,13 @@
   `(with-errors-being-passed-on ~transport ~msg
      (transport/send ~transport (response-for ~msg ~(apply hash-map kv)))))
 
+(defn serialize-response [{:keys [serialize-format] :as msg} response]
```
Comment (expez) [inline src/refactor_nrepl/middleware.clj]: This should probably be `serialization-format`

**Comment by expez** [comment]
Fix the key name and squash the commits and we should be good to go.

I'm fine with no tests / documentation while you figure out if this is something that's going to give you any value at all.

**Comment by clumsyjedi** [comment]
@expez cool, thanks for the feedback. Key has been renamed, commits have been squashed.

**Comment by expez** [comment]
Great, thanks! :+1:

**Pull request merged**